apps/prairielearn/src/sync/course-db.ts

```diff
@@ -404,15 +404,30 @@ export async function loadFullCourse(
   perf.end('loadQuestions');
   const courseInstanceInfos = await loadCourseInstances(courseDir);
   const courseInstances: Record<string, CourseInstanceData> = {};
-  for (const courseInstanceId in courseInstanceInfos) {
-    // TODO: is it really necessary to do all the crazy error checking on `lstat` for the assessments dir?
-    // If so, duplicate all that here
-    const assessments = await loadAssessments(courseDir, courseInstanceId, questions);
-    const courseInstance = {
+  for (const [courseInstanceId, courseInstance] of Object.entries(courseInstanceInfos)) {
+    const courseInstanceAccessibleNowOrInFuture = (courseInstance.data?.allowAccess ?? []).some(
+      (rule) => {
+        const startDate = rule.startDate ? parseAllowAccessDate(rule.startDate) : null;
+        const endDate = rule.endDate ? parseAllowAccessDate(rule.endDate) : null;
+
+        if (startDate && isFuture(startDate)) return true;
+        if (endDate && isFuture(endDate)) return true;
```

Comment:
That's a great point; no it's not on purpose. I changed this around in a840eed and inverted all the logic so that it computes "is it expired" instead of "is it not expired". I'd appreciate a check to make sure that looks good.